pkg/oauth/provider.go

```diff
@@ -0,0 +1,33 @@
+package oauth
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/alexandrevilain/postgrest-auth/pkg/model"
+)
+
+// Oauth2Payload is payload struct to retrive from provider login
+type Oauth2Payload struct {
+	State string `json:"state"`
+	Token string `json:"token"`
+}
+type provider struct {
+}
+
+// GenreatePassword generate random password
+func GenreatePassword(length int) string {
```

Comment:
This function isn't provider-related. I think you can move it to the user's methods.